### DISCUSSION TITLE
How to add a border around spaces?

### DISCUSSION THREAD

**Discussion by jay0ma** [FelixKratz/SketchyBar#299] [Q&A] [answered]
How to add a border around spaces?

**Comment by jay0ma** [comment]
![Screenshot 2022-12-12 at 5 33 17 PM](https://user-images.githubusercontent.com/71800980/206998322-5a8992ff-5c1d-40c0-89f6-bbfe484dae13.png)
I have my spaces bar setup like this, and I want a white border around it. Is there a way I could do that?
![Screenshot 2022-12-12 at 5 34 10 PM](https://user-images.githubusercontent.com/71800980/206998506-2787fbff-fc04-4c93-b79e-a205baddda8b.png)
This is my code right now

**Reply by FelixKratz** [reply]
I found a bug with negative background paddings and brackets, should be fixed in: https://github.com/FelixKratz/SketchyBar/commit/277b253c4d3d2d15ccdd90936a02cd649ef89b35
Apart from that you need to disable the background of the spaces themselves, because otherwise their background occludes the bracket background.

For the next release I am aiming to rework the internal bracket logic quite a bit #300.
Until this release I would advise not to use brackets for spaces.

**Reply by jay0ma** [reply]
Alright thanks!

**Reply by FelixKratz** [reply]
I did the first part of the rework and it should work perfectly fine with brackets now on latest master: https://github.com/FelixKratz/SketchyBar/commit/bb85f7640e32e74a3fb78a4c00e97b7a0a43baee

**Comment by FelixKratz** [comment, marked as answer]
With the newest release this is possible with brackets, see this example:

```bash
sketchybar --add bracket spaces '/space\..*/'           \
           --set spaces  background.color=$BACKGROUND_1 \
                         background.border_color=$BACKGROUND_2 \
                         background.border_width=2 \
                         background.height=30 \
                         background.drawing=on
```
make sure to disable the background of the spaces themselves, else the bracket might be hidden behind it.
